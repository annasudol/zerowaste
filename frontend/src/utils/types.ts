export interface LocationTypes {
    hash?: string
    pathname: string
    search?: string
    state: {
        ingredients?: string[]
        backPath?: string
        callRefetch?: boolean
    };
    key?: string
}



export interface ListType {
    title: string
    id: number
}


export interface RecipeStateProps {
    title: string
    servings: number
    image: string
    readyInMinutes: number
    ingredients: string[] | []
    detailedIngredients: string[] | []
    instructions: string
    sourceUrl: undefined | string
}