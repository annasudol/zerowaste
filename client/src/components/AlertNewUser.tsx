import * as React from 'react';
import { useSubscription } from '@apollo/react-hooks';
import gql from 'graphql-tag';


const NEW_USER_SUBSCRIPTION = gql`
  subscription userCreated {
    userCreated {
        name
        email
    }
}
`;

export const AlertNewUser: React.FunctionComponent = (): React.ReactElement => {
  const { data, loading } = useSubscription(NEW_USER_SUBSCRIPTION);
  console.log(data, 'data')
  return <div>hello</div>
}