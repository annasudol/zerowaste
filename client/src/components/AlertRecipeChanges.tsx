// import * as React from 'react';
// import { useSubscription } from '@apollo/react-hooks';
// import gql from 'graphql-tag';

// const GET_USER_SUBSCRIPTION = gql`
//   subscription changesInRecipe {
//     changesInRecipe {
//     id
//     title
//     ingredients
//   }
//   }
// `;


// export const AlertRecipeChanges: React.FC = (): React.ReactElement | null => {
//   const { data } = useSubscription(GET_USER_SUBSCRIPTION);

//   if (data) {
//     console.log(data, "hello")
//   }
//   return null
// }