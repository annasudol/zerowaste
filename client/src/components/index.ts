export * from './App';
export * from './Button';
export * from './ErrorMessage';
export * from './Image';
export * from './LoadingBar';
export * from './LoginSignUp';
export * from './Navigation';
export * from './RecipesList';
export * from './RecipeDetails';
export * from './RecipeForm';
export * from './RecipeItem';
export * from './SearchRecipesPage';
export * from './SearchRecipesForm';