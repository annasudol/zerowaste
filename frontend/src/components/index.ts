export * from './app/App';
export * from './loginSignUpForm/LoginSignUpForm';
export * from './autoCompleteForm/AutoCompleteForm';
export * from './dialogDeleteRecipe/DialogDeleteRecipe';
export * from './errorMessage/ErrorMessage';
export * from './image/Image';
export * from './imageUpload/ImageUpload';
export * from './list/List';
export * from './protectedRoute/ProtectedRoute';
export * from './loadingBar/LoadingBar';
export * from './listInput/ListInput';
export * from './navigation/Navigation';
export * from './recipeItem/RecipeItem';
export * from './recipeForm/RecipeForm';
export * from './searchRecipesForm/SearchRecipesForm';