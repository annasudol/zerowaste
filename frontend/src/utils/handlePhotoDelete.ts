export const handlePhotoDelete = async (url: string) => {
    const publicId = url.match(/(?=).{20}?(?=.jpe?g|.png)/);
    if (publicId) {
        try {
            const response = await fetch(`http://localhost:${process.env.REACT_APP_PORT}/delete/`, {
                method: 'POST',
                headers: {
                    'Content-Type': 'application/json'
                },
                body: JSON.stringify({ public_id: publicId[0] })
            });
            console.log(response, "response")
            return response.json();

        } catch { }
    }

}