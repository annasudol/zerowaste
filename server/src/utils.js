
module.exports.paginateResults = ({
  page = 0,
  pageSize = 10,
  results,
}) => {
  const resultsIntoPages = page * pageSize;
  return results.slice(resultsIntoPages, resultsIntoPages + pageSize);
};
