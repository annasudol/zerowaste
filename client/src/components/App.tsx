import * as React from "react";
import { Navigation } from "../components";
import { Login, SignUp, Main, RecipesList, AddRecipeForm, RecipeInfo, UserInfo } from "../pages";

import { AppRoutes } from '../../routes';
import { Route, Switch } from 'react-router-dom';
import { ErrorMessage } from './ErrorMessage';


export const App: React.FunctionComponent = () => {
    const token = window.localStorage.getItem('token');

    return (
        <>
            <Navigation token={token} />
            <Switch>
                <Route path={AppRoutes.Home} component={Main} exact />
                <Route path={AppRoutes.RecipesList} component={RecipesList} />
                <Route path={AppRoutes.Recipe} component={RecipeInfo} />

                {!token ? (
                    <>
                        <Route path={AppRoutes.Login} component={Login} />
                        <Route path={AppRoutes.SignUp} component={SignUp} />
                    </>
                ) : (
                        <>
                            <Route path={AppRoutes.User} component={UserInfo} />
                            <Route path={AppRoutes.AddRecipe} component={AddRecipeForm} />
                        </>
                    )}
                <Route path="*">
                    <ErrorMessage message="page not found" />
                </Route>
            </Switch>
        </>
    )
}