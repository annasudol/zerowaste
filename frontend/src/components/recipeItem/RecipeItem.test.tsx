import { shallow } from 'enzyme';
import React from 'react';
import { RecipeItem, GET_RECIPE_DETAILS } from '../../components';
import toJson from 'enzyme-to-json';
import { renderApollo, cleanup,  fireEvent, waitForElement } from '../../test.utils';
import { HashRouter } from 'react-router-dom';
import { act } from "react-dom/test-utils";



describe('component', (): void => {
  describe('RecipeItem', (): void => {
    afterEach(cleanup);
    it('match snapshot', (): void => {
      const wrapper = shallow(<RecipeItem  id="1" title="test" image="img" ingredients={["ing1"]}/>);
      expect(toJson(wrapper)).toMatchSnapshot();
    });
    it('responds for opening delete recipe modal', (): void => {
      const wrapper = shallow(<RecipeItem  id="1" title="test" image="img" ingredients={["ing1"]} deleteEditBtn/>);
      act(()=> {
        wrapper.find('Button[className="mr-2"]').simulate('click');
      })
      expect(toJson(wrapper)).toMatchSnapshot();

    });
    //TODO: finish test
    it('match snapshot 2', async () => {

        const mocks = [
            {
              request: {
                query: GET_RECIPE_DETAILS,
                variables: { id: "5ef248ca324f6aaa02bfd7a0" },
                results: { recipe:  {
                    _typename: "RecipeDetails",
                    author: null,
                    detailedIngredients: ['test'],
                    image: "http://res.cloudinary.com/drgb4slzt/image/upload/v1593443213/belhdjaq4pevmwydoept.jpg",
                    instructions: "test",
                    readyInMinutes: 130,
                    servings: 8,
                    sourceUrl: "myblog.com",
                    title: "Apple Pie",
                    user: {
                        __typename: "User",
                        name: "anna"
                    }
                }},
              }
            }
          ]
          const {findByText} = renderApollo(<HashRouter><RecipeItem  id="5ef248ca324f6aaa02bfd7a0" title="test" image="img" ingredients={["ing1"]} deleteEditBtn/></HashRouter>, {mocks})
          const editButton =  await waitForElement(() => findByText(/edit Recipe/i));
          act(()=> {
            fireEvent.click(editButton);
          })
      });
    });
});

