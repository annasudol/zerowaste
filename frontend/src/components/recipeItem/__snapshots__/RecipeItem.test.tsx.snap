// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component RecipeItem match snapshot 1`] = `
<div
  className="flex mb-4 mr-4 max-w-sm lg:max-w-xl mb-3 list--item relative"
  key="1"
>
  <div
    className="w-42"
  >
    <Image
      alt="test"
      className="rounded-m m-4"
      size="small"
      src="img"
    />
  </div>
  <div
    className="mt-4 mb-4"
  >
    <h3
      className="font-bebas text-lightGreen mb-1"
    >
      test
    </h3>
    <span
      className="font-roboto uppercase text-xs font-semibold text-darkGray"
    >
      ingredients: 
    </span>
    <span
      className="font-roboto text-xs"
    >
      ing1
    </span>
    <br />
    <Link
      className="coral-link"
      to="/recipe/1"
    >
      see more
    </Link>
  </div>
</div>
`;

exports[`component RecipeItem responds for opening delete recipe modal 1`] = `
<DialogDeleteRecipe
  image="img"
  open={true}
  recipeId="1"
  title="test"
  toggleOpen={[Function]}
/>
`;
